### PR TITLE
spv-out: support version 1.4

### PR DESCRIPTION
**Pull request by kvark**
Looks like 1.4 expects all global variables to be a part of the interface:
> Starting with version 1.4, the interface’s [storage classes](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Storage_Class) are all storage classes used in declaring all global variables referenced by the entry point’s call tree.

### CODE REVIEW

File: src/back/spv/writer.rs [hunk 1]
```diff
@@ -585,6 +585,12 @@ impl Writer {
             }
 
             let mut gv = self.global_variables[handle.index()].clone();
+            if let Some(ref mut iface) = interface {
+                // Have to include global variables in the interface
+                if self.physical_layout.version >= 0x10400 {
```
Comment (teoxoy) [inline src/back/spv/writer.rs:590]: Would be nice to do the version conversion later and keep the tuple, but can be another PR.

https://github.com/gfx-rs/naga/blob/c7d02151f08d6285683795289b5725b827d836d1/src/back/spv/writer.rs#L48

**Review by teoxoy — CHANGES_REQUESTED** [review]

File: src/back/spv/writer.rs (hunk 1 above)
Comment (kvark) [reply src/back/spv/writer.rs:590]: possible, but not very important. We are already checking for the raw version in another place, so the PR is consistent with the code.

File: tests/snapshots.rs [hunk 2]
```diff
@@ -561,6 +561,7 @@ fn convert_wgsl() {
             "workgroup-var-init",
             Targets::WGSL | Targets::GLSL | Targets::SPIRV | Targets::HLSL | Targets::METAL,
         ),
+        ("sprite", Targets::SPIRV),
```
[the remark this answers is not in the document]
Comment (kvark) [reply tests/snapshots.rs:564]: trying to use git on windows, and all the generated files are marked as modified after cargo test :(
added the output now

**Comment by kvark** [comment]
@teoxoy unfortunately the test validation is blocked on a thing. Our `make validate-spv` has "vulkan1.0" environment hard-coded currently, so it doesn't pass validation. We need to select the environment based on the SPV version that we generate. We could do this, for example, by changing the naming scheme ("xxx.spv14asm"), or by having sub-folders inside, e.g "tests/out/spv/1.4".

**Comment by teoxoy** [comment]
I think it would be feasible to read the version from the file and map it to a vulkan version but that sounds quite involved.
We can go with a sub-folder for now - up to you.

**Comment by kvark** [comment]
I think we should just merge this with some sort of a workaround, not blocking for a general solution (be it subfolders or not). Even without a test would be fine, since it's not a regression.

**Comment by kvark** [comment]
@teoxoy PR now comes with the general solution that extracts SPV version from the comment

**Review by teoxoy — APPROVED** [review]
> PR now comes with the general solution that extracts SPV version from the comment

That's great, thanks for addressing it here!

**Pull request merged**